### PR TITLE
Update deprecation tags for basic auth env vars for agent

### PR DESCRIPTION
**Pull request by michel-laterman**
Some vars are replaced by service_token, some are removed, some have the behaviour changed

### CODE REVIEW

**Comment by dedemorton** [comment]
Looks like you missed one of the notes under `KIBANA_FLEET_SETUP`.

**Review by dedemorton — APPROVED** [review]
Approving conditionally (with the assumption that you will replace the deprecation statement under KIBANA_FLEET_SETUP.

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 1]
```diff
@@ -110,6 +112,23 @@ When set to `1`, this automatically forces {fleet} enrollment as well.
 
 // =============================================================================
 
+// tag::fleet-setup[]
+
+[id="env-{type}-fleet-setup"]
+`FLEET_SETUP`
```
Comment (michel-laterman) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:118]: @dedemorton,  we were actually missing `FLEET_SETUP`. I've added it with the deprecation notice

**Comment by dedemorton** [comment]
@michel-laterman Are you going to backport this to 7.16? I _think_ the backport automation only works when the PR is in main.

**Comment by michel-laterman** [comment]
yeah, i'll backport

**Pull request merged**